stdlib/LinearAlgebra/test/diagonal.jl

```diff
@@ -178,6 +179,19 @@ Random.seed!(1)
         qrA = qr(A)
         @test qrA \ D ≈ A \ D
 
+        # HermOrSym
+        A     = rand(elty, n, n)
+        Asym  = Symmetric(A + transpose(A), :U)
+        Aherm = Hermitian(A + adjoint(A), :U)
+        @test Array(D*transpose(Asym)) ≈ Array(D) * Array(transpose(Asym))
```

Comment:
Which method is this supposed to test?

Reply:
[This one](https://codecov.io/gh/julialang/julia/src/master/stdlib/LinearAlgebra/src/diagonal.jl#L290)

Reply:
Hm. Did you verify that this actually exercises the method? I don't think these methods are ever called by anything. I think it would be worth simply deleting the methods instead of trying to test them.

Reply:
😄 , @andreasnoack, I think I was asking about those methods, and you suggested they are there for disambiguation. It could be that these methods are only reached when you call the wrapper `Transpose` explicitly, and not just `transpose`. The latter may not do anything to `Symmetric` matrices, while the first may simply wrap a layer around? So it could well be that the line in question does get called if one replaces the `transpose`s by `Transposes`.

Reply:
>  I think I was asking about those methods, and you suggested they are there for disambiguation.

Ha. It's very likely. Do you remember if you tried to remove them? I don't think it's worth having these unless they fix an ambiguity error.

Reply:
No, I haven't tried since I wasn't involved directly. That issue came up here https://github.com/JuliaLang/julia/pull/31400#pullrequestreview-217126405.

Reply:
I just tried this and the method @andreasnoack pointed out is not ever called.

Reply:
I'll try with the explicit wrapper too

Reply:
I'm not quite sure why one would explicitly want a `Transpose` wrapper of a `Symmetric` matrix, but maybe this is somewhat useful in block matrices, where symmetry is assessed on the block level.

Reply:
Maybe the person writing the code doesn't know the matrix is `Symmetric`?

Reply:
Personally, I just never use `Transpose`, but `transpose`, and that dispatches to a no-op for `Symmetric` matrices. That person would need to call the wrapper explicitly in the (possibly generic) code, instead of the "give me the transpose and decide yourself how to do it" `transpose`.

Reply:
I mean, I do what you do also. But people have the desire to all kinds of weird stuff.

Reply:
I think we should just delete the methods. It is very unlikely that anybody would ever end up with `Adjoint`-wrapped `Hermitian`s and I don't think we should accommodate that. If anybody ends up with one of these matrices then they'll just hit a slow fallback which I think is fine.

Reply:
If that's harmless in terms of ambiguity, I agree.